medusa/providers/nzb/newznab.py

```diff
@@ -324,6 +331,42 @@ def image_name(self):
             return self.get_id() + '.png'
         return 'newznab.png'
 
+    def _match_indexer(self):
+        """Use the indexers id and externals, and return the most optimal indexer with value.
+
+        For newznab providers we prefer to use tvdb for searches, but if this is not available for shows that have
+        been indexed using an alternative indexer, we could also try other indexers id's that are available
+        and supported by this newznab provider."""
+        # The following mapping should map the newznab capabilities to our indexers or externals in indexer_config.
+        map_caps = {INDEXER_TMDB: 'tmdbid', INDEXER_TVDBV2: 'tvdbid', INDEXER_TVMAZE: 'tvmazeid'}
+
+        return_mapping = {}
+
+        if self.cap_tv_search == 'True':
+            # We didn't get back a supportedParams, lets return, and continue with doing a search string search.
+            return {}
+
+        for search_type in self.cap_tv_search.split(','):
+            if search_type == 'tvdbid':
```

Comment:
Maybe do `if search_type == 'tvdbid' and self._get_tvdb_id():`
for when there isn't a tvdbid but the indexer supports it